store/src/main/java/org/apache/rocketmq/store/schedule/ScheduleMessageService.java

```diff
@@ -431,35 +492,414 @@ public void executeOnTimeup() {
             ScheduleMessageService.this.deliverExecutorService.schedule(new DeliverDelayedMessageTimerTask(this.delayLevel,
                 failScheduleOffset), DELAY_FOR_A_WHILE, TimeUnit.MILLISECONDS);
         }
+    }
+
+    class AsyncDeliverDelayedMessageTimerTask implements Runnable {
```

Comment:
`AsyncDeliverDelayedMessageTimerTask` is similar with `DeliverDelayedMessageTimerTask`, consider extending `DeliverDelayedMessageTimerTask`